### ISSUE TITLE
unique death normalization improvements

### ISSUE THREAD

**Issue by bhaak** [junethack/Junethack#43]
Look at the proposed unique death normalization from these two posts:
http://mathematicalcoffee.blogspot.com.au/2014/07/junethack-2014-is-over.html     http://74.135.83.0:8018/nethack-stuff/junethack-unique-deaths-comments-2014-jonadab.html

**Comment by mathematicalcoffee** [comment]
FYI #48 pull request implements some of the widely-agreed-upon death normalisations.

**Comment by FredrIQ** [comment]
Minions of deities can be polymorphed, this is not taken into account since the list is hardcoded. I suggest hardcoding deity names instead, which would also catch aligned/high priests.

**Comment by RojjaCebolla** [comment]
Some death messages seem to lack a right paren, perhaps it should be made optional in the relevant regex?

```
slipped while mounting a monster (a saddled speedhorse
killed by a monster (Tasmanian devil
```

**Comment by actual-nh** [comment]
Are things like "moderately wounded" ("killed by a moderately wounded goblin"), "seriously wounded", and "uninjured" supposed to be removed for unique deaths?

**Comment by bhaak** [comment]
Vanilla has removed the wouded state from the death message for the time being and I think every variant that had it before has upgraded already.
So there's is currently no pressing issue to look at that. But I guess it should be removed if it ever comes back. Otherwise you just have uncesseray grinding.

**Comment by bhaak** [comment]
Vanilla has removed the wouded state from the death message for the time being and I think every variant that had it before has upgraded already.
So there's is currently no pressing issue to look at that. But I guess it should be removed if it ever comes back. Otherwise you just have uncesseray grinding.

**Comment by actual-nh** [comment]
> Vanilla has removed the wounded state from the death message for the time being and I think every variant that had it before has upgraded already.
> So there's is currently no pressing issue to look at that. 

I was quoting from the current Junethack "unique deaths" listing, so there's at least one version still putting it in. (There aren't very many so far, but it's only June 2nd...)

> But I guess it should be removed if it ever comes back. Otherwise you just have uncesseray grinding.

Agreed.

**Comment by actual-nh** [comment]
There are also a few cases in the current listing where the trimming seems to have gone a bit too far:
```
9 | imp
9 | killed by a fox
9 | orc
```
Additional ones include demon, hobgoblin, rock mole...

**Comment by bhaak** [comment]
> There are also a few cases in the current listing where the trimming seems to have gone a bit too far:

No, those are from 1.3d games: https://junethack.net/server/hdf_13d

This old version of NetHack didn't specify how the monster killed you.

**Comment by actual-nh** [comment]
> > There are also a few cases in the current listing where the trimming seems to have gone a bit too far:
> 
> No, those are from 1.3d games: https://junethack.net/server/hdf_13d
> 
> This old version of NetHack didn't specify how the monster killed you.

OK; changing such to "killed by [a|an] <foo>" would be nice in the long run. (Any idea what version still has wounded state?)

**Comment by bhaak** [comment]
Without checking the code, I'm pretty sure that the wounded descriptions were a 3.7 thing.
xNetHack is the only variant that follows 3.7 closely, so it's probably from that.

But that also means I still have to fix that.

**Comment by actual-nh** [comment]
I can work on regexes.

**Comment by RojjaCebolla** [comment]
object materials are also showing up in deaths. harder but not impossible to grind
https://imgur.com/2qwRQZK.png

**Comment by actual-nh** [comment]
> object materials are also showing up in deaths. harder but not impossible to grind
> 
> https://imgur.com/2qwRQZK.png

Should `silver` be removed?

**Comment by actual-nh** [comment]
"poisoned by" -> "killed by":
```
1 | poisoned by Pestilence (with the Amulet)
1 | poisoned by a bone orcish arrow
1 | poisoned by a giant mosquito
1 | poisoned by a killer bee
1 | poisoned by a little dart
1 | poisoned by a noxious patch
1 | poisoned by a quasit
```
